metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlTagService.java

```diff
@@ -307,25 +325,31 @@ public void remove(final QualifiedName name, final Set<String> tags, final boole
         @Nullable final Set<String> excludeTags,
         @Nullable final String sourceName,
         @Nullable final String databaseName,
-        @Nullable final String tableName
+        @Nullable final String tableName,
+        @Nullable final QualifiedName.Type type
     ) {
         Set<String> includedNames = Sets.newHashSet();
         final Set<String> excludedNames = Sets.newHashSet();
-        final String wildCardName = QualifiedName.toWildCardString(sourceName, databaseName, tableName);
+        final String wildCardName =
+            MySqlServiceUtil.qualifiedNameToWildCardQueryString(sourceName, databaseName, tableName);
         //Includes
         final Set<String> localIncludes = includeTags != null ? includeTags : Sets.newHashSet();
         try {
-            String query
-                = String.format(QUERY_SEARCH, "in ('" + Joiner.on("','").skipNulls().join(localIncludes) + "')");
+            StringBuilder query = new StringBuilder(
+                String.format(QUERY_SEARCH, "in ('" + Joiner.on("','").skipNulls().join(localIncludes) + "')"));
             final Object[] params = {localIncludes.size() == 0 ? 1 : 0, wildCardName == null ? 1 : 0, wildCardName};
-            includedNames.addAll(jdbcTemplate.query(query, params,
+            addTypeLimitToTagQuery(query, type);
```

Comment:
We should add it to the query as a parameter like 1=? or name rlike ?

Reply:
I feel it's hard to understand the logic of "a parameter like 1=?" which isn't needed,  and prefer making it explicitly to add the query if needed.

Reply:
Utilizing PreparedStatement approach correctly.

Reply:
Sounds right.